src/platforms/common/swdptap.c

```diff
@@ -53,6 +54,20 @@ void swdptap_init(void)
 	swd_proc.seq_out_parity = swdptap_seq_out_parity;
 }
 
+/* Busy-looping delay snippet for GPIO bitbanging (rely on inlining) */
+static inline void platform_delay_busy(const uint32_t loops)
+{
+#if 1
+	register uint32_t i = loops;
+	do {
+		__asm__("nop");
+	} while (--i > 0U);
+#else
+	for (register uint32_t counter = loops; counter > 0; --counter)
```

Comment:
Same comments as in jtagtap.c

Reply:
Updated the delay loop to `--counter > 0U;)` form.